courses/management/commands/manage_certificates.py

```diff
@@ -0,0 +1,189 @@
+        run = options.get("run")
+
+        if not (revoke or unrevoke) and not create:
+            raise CommandError(
+                "The command needs a valid action e.g. --revoke, --unrevoke, --create"
+            )
+        try:
+            user = fetch_user(options["user"]) if options["user"] else None
+        except User.DoesNotExist:
+            user = None
+
+        # A run is needed for revoke/un-revoke and certificate creation
+        if not run:
+            raise CommandError("The command needs a valid course run")
+
+        # Unable to obtain a run object based on the provided courseware id
+        try:
+            course_run = CourseRun.objects.get(courseware_id=options["run"])
+        except CourseRun.DoesNotExist:
+            raise CommandError(
+                "Could not find run with courseware_id={}".format(options["run"])
+            )
+
+        # Handle revoke/un-revoke of a certificate
+        if revoke or unrevoke:
+            if not user:
+                raise CommandError("Revoke/Un-revoke operation needs a valid user")
+
+            revoke_status = manage_course_run_certificate_access(
+                user=user,
+                courseware_id=course_run.courseware_id,
+                revoke_state=True if revoke else False,
+            )
+
+            if revoke_status:
+                self.stdout.write(
+                    self.style.SUCCESS(
+                        "Certificate for {} has been {}".format(
+                            "run: {}".format(run), "revoked" if revoke else "un-revoked"
+                        )
+                    )
+                )
+            else:
+                self.stdout.write(self.style.WARNING("No changes made."))
+
+        # Handle the creation of the certificates
+        elif create:
+            override_grade = options.get("grade")
+
+            if override_grade and (override_grade < 0.0 or override_grade > 1.0):
+                raise CommandError("Invalid value for grade. Allowed range: 0.0 - 1.0")
+
+            if override_grade and not user:
+                raise CommandError(
+                    "Override grade needs a user (The grade override operation is not supported for multiple users)"
+                )
+
+            # If user=None, Grades for all users in this runs will be fetched
+            edx_grade_user_iter = get_edx_grades_with_users(course_run, user=user)
+
+            results = []
+            is_grade_override = override_grade is not None
+            for edx_grade, user in edx_grade_user_iter:
+                (
+                    course_run_grade,
+                    created_grade,
+                    updated_grade,
+                ) = ensure_course_run_grade(
+                    user=user,
+                    course_run=course_run,
+                    edx_grade=edx_grade,
+                    should_update=True,
+                )
+
+                if is_grade_override:
```

Comment:
Could we move this out of the `-create` option into it's own option? something like `./manage.py manage_certificates -update_grade ...`

Reply:
Good suggestion but let me add some more context. So, most of this command is inspired by [this](https://github.com/mitodl/mitxpro/blob/master/courses/management/commands/sync_grades_and_certificates.py) and [this](https://github.com/mitodl/mitxpro/blob/master/courses/management/commands/revoke_certificate.py) in xPRO.

A use case: Let's say we want to create a certificate for a user specifically and also want to override the grade for that along the way without respecting what grades the user might have in edX, we will need to pass the `-grade` value because we won't know what grading policy has been set in edX. Since this is still a create operation but with grade we can use it with `-create`.

Since you mentioned, I though,t or something, How about when we just want to override/update the grade and not create the certificate we have no option for that in command. So, I create `-update_grade` operation which will just update the grade of that user in the database and not touch the certificate. What would you say about that? @annagav

Reply:
@annagav I've addressed the feedback as per my above message, Of course, let me know if you think otherwise and I'll update the PR accordingly.